lib/pirate/service/utilities.rb

```diff
@@ -0,0 +1,24 @@
+module Pirate
+  module Service
+    module Utilities
+      # Utility method for either returning a non-empty collection
+      # or returning nil
+      #
+      # @param [Enumerable] the collection to test
+      # @return [Enumerable, Nil]
+
+      def unless_empty(collection)
+        return collection unless empty_or_nil?(collection)
+      end
+
+      # Shorthand method for checking if something is empty or nil
+      #
+      # @param [Object] value any value to test against
+      # @return [Truthy]
+
+      def empty_or_nil?(value)
+        !!(value.nil? || value&.empty?)
```

Comment:
This is called the "safe navigation operator", which is basically the equivalent of `#try`. Though, since `#try` is implemented in Rails and this lib doesn't depend on Rails, I went something that's available in Ruby (2.3.0 onwards).

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/